test/fake_report_server.ts

```diff
@@ -0,0 +1,97 @@
+import { Server, Socket } from 'net'
+import express from 'express'
+import { pipeline, Writable } from 'stream'
+import http from 'http'
+import { promisify } from 'util'
+
+type Callback = (err?: Error | null) => void
+
+/**
+ * Fake implementation of the same report server that backs Cucumber Reports
+ * (https://messages.cucumber.io). Used for testing only.
+ */
+export default class FakeReportServer {
+  private readonly sockets = new Set<Socket>()
+  private readonly server: Server
+  private receivedBodies = Buffer.alloc(0)
+  public receivedHeaders: http.IncomingHttpHeaders = {}
+
+  constructor(private readonly port: number) {
+    const app = express()
+
+    app.put('/s3', (req, res) => {
+      this.receivedHeaders = { ...this.receivedHeaders, ...req.headers }
+
+      // eslint-disable-next-line @typescript-eslint/no-this-alias
+      const server = this
+      const captureBodyStream = new Writable({
+        write(chunk: Buffer, encoding: string, callback: Callback) {
+          server.receivedBodies = Buffer.concat([server.receivedBodies, chunk])
+          callback()
+        },
+      })
+
+      pipeline(req, captureBodyStream, (err) => {
+        if (err !== null && err !== undefined)
```

Comment:
We have a helper for this `doesHaveValue` in src/value_checker

Reply:
Fixed in deprecated and will be removed in the next majo

Reply:
Fixed in 9629395a0ab37f0bf59518e6c68e970d264c3360